Sources/AWSSDKSwiftCore/MetaDataService.swift

```diff
@@ -111,31 +98,101 @@ struct MetaDataService {
                 }
             }
         }
-        
+
         if let e = error {
             throw e
         }
-        
+
         return response!
     }
-    
-    func getRoleName() throws -> String {
-        let response = try request(url: URL(string: self.urlString)!, timeout: 2)
-        switch response.statusCode {
-        case 200:
-            return String(data: response.body.asData(), encoding: .utf8) ?? ""
-        default:
-            throw MetaDataServiceError.couldNotGetInstanceRoleName
+
+    struct MetaData {
```

Comment:
is `MetaData` able to adopt Codable? If so I think it's better for maintenance.
(This is just my opinion.)